facilitator/src/task.rs

```diff
@@ -0,0 +1,91 @@
+mod pubsub;
+
+use anyhow::{Context, Result};
+use serde::Deserialize;
+use std::fmt::Debug;
+
+pub use pubsub::PubSubTaskQueue;
+
+/// Represents a task that can be assigned to a worker
+pub trait Task: Debug + Sized {
+    /// Attempts to deserialize a task structure from the provided JSON string
+    fn from_json(json: &[u8]) -> Result<Self>;
+}
```

Comment:
Like in Go, I think you can get want you want from pre-existing types:

```rust
pub trait Task: Debug + Sized + serde::de::DeserializeOwned {
}
```

Then you can take advantage of `from_reader`: https://docs.serde.rs/serde/de/trait.DeserializeOwned.html

And `&[u8]` implements Read: https://doc.rust-lang.org/std/primitive.slice.html#impl-Read